### ISSUE TITLE
Centering and Sharpening balance

### ISSUE THREAD

**Issue by JaejinCho** [facebookresearch/dino#32]
Hello, thanks for sharing your great work with codes!

I was wondering if you experimented with no sharpening and no centering at all? I was thinking if using either one causes some collapse, why not use none of them during training.

Thanks,
Jaejin Cho

**Comment by mathildecaron31** [comment]
Hi @JaejinCho 
Thanks for your interest and your kind words.

When neither sharpening or centering is applied the model collapses to one dominant dimension, exactly in the same way as when only sharpening is applied.

**Comment by arrufat** [comment]
Sorry to ask this on a closed issue, but I have some trouble understanding the effect of the centering, since the Softmax function is translation invariant. Actually, Softmax is usually [implemented](https://github.com/pytorch/pytorch/blob/4cb534f92ef6f5b2ec99109b0329f93a859ae831/aten/src/ATen/native/SoftMax.cpp#L44) with by subtracting the maximum value to the inputs (which can be seen as some kind of centering) for numerical stability.

Am I missing something?

Thanks in advance for your reply, the paper was a fascinating read :)
